### DISCUSSION TITLE
How to use static files in server side code?

### DISCUSSION THREAD

**Discussion by manuganji** [sveltejs/kit#10543] [Q&A]
I have a few `.mjml` templates that I use with `nodemailer` and `nodemailer-mjml` plugins. However they are not packaged into the final build and they are not available to the function that uses them.

How do I prevent these files from getting stripped out of server build?

I tried putting them in `static` folder but then the files ended up in the client side folder, again unavailable to the server side function.

I believe using some static files other than html, css and js is something fundamental. It's a unilateral decision to strip everything out without giving the developer an opt-out mechanism.

**Comment by teemingc** [comment]
You should be able to import it in your server code and this will be stored as a server asset.

https://vitejs.dev/guide/assets.html#static-asset-handling

> Common image, media, and font filetypes are detected as assets automatically. You can extend the internal list using the assetsInclude option.

You may have to modify your vite config to get it to work with .mjml files

https://vitejs.dev/config/shared-options.html#assetsinclude

**Reply by manuganji** [reply]
I tried this before asking and it didn't work. I'll try to provide a reproducible repo for your testing.

**Reply by teemingc** [reply]
similar issue https://github.com/sveltejs/kit/issues/10594
